core/processor/abci.go

```diff
@@ -575,14 +575,6 @@ func (app *App) OnInitChain(req tmtypes.RequestInitChain) tmtypes.ResponseInitCh
 	ctx = vgcontext.WithTraceID(ctx, hash)
 	app.blockCtx = ctx
 
-	// Start the snapshot engine letting it clear any pre-existing state so that if we are
```

Comment:
Is that not applicable anymore if we decide to replay explicitly even if we have local snapshots?

Reply:
It is, I've just moved it here:
```
	if startHeight == 0 {
		// starting a new chain or replaying, not loading snapshot
		// forced chain replay, we need to remove all old snapshots and start again
		e.initialised = false
		e.db.Close()
		e.ClearAndInitialise()
		return false, nil
	}
```

Reply:
Makes senses, I was too impatient to finish reading it.

Reply:
Although I'm no longer checking for the error! I'll fix that.

Reply:
fair enough.